examples/1.0.0/bnpl-openapi.yaml

```
@@ -0,0 +1,457 @@
+      operationId: getAuthorization
+      parameters:
+        - name: AuthorizationToken
+          in: query
+          description: Authorization token value elicited from loan initiation endpoint
+          required: true
+          schema:
+            $ref: '#/components/schemas/AuthorizationToken'
+      responses:
+        "302":
+          description: Instruction to redirect End User, based on validation of the authorisation token
+          headers:
+            Location:
+              description: URL to which the customer is redirected
+              schema:
+                type: string
+        default:
+          $ref: '#/components/responses/ErrorResponse'
+  /customers:
+    post:
+      summary: Create a customer
+      description: |
+        Create a customer for a BNPL loan if they are eligible for the loan in question.
+
+        If a customer is eligible a customer resource is created, a 201 returned, and the a link to the customer resource returned.
+
+        If the customer is not eligible a 200 is returned and a reason code indicating why the customer was rejected.
+      operationId: createCustomer
+      requestBody:
+        description: The customer properties
+        content:
+          application/json:
+            schema:
+              allOf:
+                 - $ref: '#/components/schemas/CustomerProperties'
+                 - description: Terms and conditions have been reviewed and accepted
+                   type: object
+                   required:
+                     - termsAndConditionsAccepted
+                   properties:
+                     termsAndConditionsAccepted:
+                      type: boolean
+      responses:
+        "200":
+          description: Customer is not eligible for BNPL loan
+          content:
+            application/json:
+              schema:
+                type: object
+                required:
+                  - reasonCode
+                properties:
+                  reasonCode:
+                    type: string
+                additionalProperties: false
+        "201":
+          description: Customer resource has been created and can be linked to loan transaction
+          content:
+            application/json:
+              schema:
+                type: object
+                required:
+                  - customerId
+                  - links
+                properties:
+                  customerId:
+                    description: Unique identifier for the newly created customer resource
+                    type: string
+                  links:
+                    type: object
+                    required:
+                      - self
+                    properties:
+                      self:
+                        description: URL identifying this resource
+                        allOf:
+                          - $ref: '#/components/schemas/CustomerUri'
+                    additionalProperties: false
```

in OAS 3.0.x, siblings are not allow with `$ref`, it must be wrapped in `allOf`

```suggestion
                    properties:
                      self:
                        allOf:
                          - description: URL identifying this resource
                          - $ref: '#/components/schemas/CustomerUri'
                    additionalProperties: false
```